src/awkward/contents/regulararray.py

```diff
@@ -981,70 +981,197 @@ def _reduce_next(
         keepdims,
         behavior,
     ):
-        out = self.toListOffsetArray64(True)._reduce_next(
-            reducer,
-            negaxis,
-            starts,
-            shifts,
-            parents,
-            outlength,
-            mask,
-            keepdims,
-            behavior,
-        )
+        branch, depth = self.branch_depth
+        nextlen = len(self) * self._size
+
+        if negaxis == depth and not branch:
```

Comment:
A minor point: keep the text of this predicate the same as for ListOffsetArray so that it's easier to search in an editor.

```suggestion
        if not branch and negaxis == depth:
```

Reply:
Agreed, and it also makes it easier to see how the two differ.